content/fundamental-redirects.js

```diff
@@ -0,0 +1,1140 @@
+  ),
+  redirect(
+    /^samples\/webgl\/sample6\/sylvester.js$/i,
+    "http://mdn.github.io/webgl-examples/tutorial/sylvester.js",
+    { permanent: true }
+  ),
+  redirect(
+    /^samples\/webgl\/sample6\/webgl-demo.js$/i,
+    "http://mdn.github.io/webgl-examples/tutorial/sample6/webgl-demo.js",
+    { permanent: true }
+  ),
+  redirect(
+    /^samples\/webgl\/sample6\/webgl.css$/i,
+    "http://mdn.github.io/webgl-examples/tutorial/webgl.css",
+    { permanent: true }
+  ),
+  redirect(
+    /^samples\/webgl\/sample7$/i,
+    "http://mdn.github.io/webgl-examples/tutorial/sample7",
+    { permanent: true }
+  ),
+  redirect(
+    /^samples\/webgl\/sample7\/cubetexture.png$/i,
+    "http://mdn.github.io/webgl-examples/tutorial/sample7/cubetexture.png",
+    { permanent: true }
+  ),
+  redirect(
+    /^samples\/webgl\/sample7\/glUtils.js$/i,
+    "http://mdn.github.io/webgl-examples/tutorial/glUtils.js",
+    { permanent: true }
+  ),
+  redirect(
+    /^samples\/webgl\/sample7\/index.html$/i,
+    "http://mdn.github.io/webgl-examples/tutorial/sample7/index.html",
+    { permanent: true }
+  ),
+  redirect(
+    /^samples\/webgl\/sample7\/sylvester.js$/i,
+    "http://mdn.github.io/webgl-examples/tutorial/sylvester.js",
+    { permanent: true }
+  ),
+  redirect(
+    /^samples\/webgl\/sample7\/webgl-demo.js$/i,
+    "http://mdn.github.io/webgl-examples/tutorial/sample7/webgl-demo.js",
+    { permanent: true }
+  ),
+  redirect(
+    /^samples\/webgl\/sample7\/webgl.css$/i,
+    "http://mdn.github.io/webgl-examples/tutorial/webgl.css",
+    { permanent: true }
+  ),
+  redirect(
+    /^samples\/webgl\/sample8$/i,
+    "http://mdn.github.io/webgl-examples/tutorial/sample8",
+    { permanent: true }
+  ),
+  redirect(
+    /^samples\/webgl\/sample8\/Firefox.ogv$/i,
+    "http://mdn.github.io/webgl-examples/tutorial/sample8/Firefox.ogv",
+    { permanent: true }
+  ),
+  redirect(
+    /^samples\/webgl\/sample8\/glUtils.js$/i,
+    "http://mdn.github.io/webgl-examples/tutorial/glUtils.js",
+    { permanent: true }
+  ),
+  redirect(
+    /^samples\/webgl\/sample8\/index.html$/i,
+    "http://mdn.github.io/webgl-examples/tutorial/sample8/index.html",
+    { permanent: true }
+  ),
+  redirect(
+    /^samples\/webgl\/sample8\/sylvester.js$/i,
+    "http://mdn.github.io/webgl-examples/tutorial/sylvester.js",
+    { permanent: true }
+  ),
+  redirect(
+    /^samples\/webgl\/sample8\/webgl-demo.js$/i,
+    "http://mdn.github.io/webgl-examples/tutorial/sample8/webgl-demo.js",
+    { permanent: true }
+  ),
+  redirect(
+    /^samples\/webgl\/sample8\/webgl.css$/i,
+    "http://mdn.github.io/webgl-examples/tutorial/webgl.css",
+    { permanent: true }
+  ),
+  // All of the remaining "samples/" URL's are redirected to the
+  // the media domain (ATTACHMENTS_AWS_S3_CUSTOM_DOMAIN).
+  redirect(
+    /^samples\/(?<sample_path>.*)$/i,
+    ({ sample_path }) =>
+      `${settings.ATTACHMENTS_AWS_S3_CUSTOM_URL}/samples/${sample_path}`,
```

Comment:
We **might** have to define a `ATTACHMENTS_AWS_S3_CUSTOM_URL` constant, although, if we do, I think we should use a different name because this has little to do with what we normally call "file attachments" on MDN. Currently for MDN, the values are https://media.prod.mdn.mozit.cloud for the `prod` environment and https://media.stage.mdn.mozit.cloud for `stage`. I used the word "**might**" because the content for these archaic `/samples/...` URL's is the same for `dev`, `stage` and `prod`, and it's frozen in time as well, so we could get by with using https://media.prod.mdn.mozit.cloud, no matter the target environment.

Reply:
Within the current English documents, I found only 13 uses of these archaic URL's:
```
content • files/en-us/web/api/document/currentscript/index.html:
  32: <p><a href="/samples/html/currentScript.html">View Live Examples</a></p>

content • files/en-us/web/api/document/mozsetimageelement/index.html:
  32: <p><a href="/samples/domref/mozSetImageElement.html" title="https://developer.mozilla.org/samples/domref/mozSetImageElement.html">View this example live</a>.</p>

content • files/en-us/web/api/document/onafterscriptexecute/index.html:
  31: <p><a href="/samples/html/currentScript.html">View Live Example</a></p>

content • files/en-us/web/api/document/onbeforescriptexecute/index.html:
  32: <p><a href="/samples/html/currentScript.html">View Live Examples</a></p>

content • files/en-us/web/api/domimplementation/createhtmldocument/index.html:
  68: <p><a href="/samples/domref/createHTMLDocument.html">View Live Examples</a></p>

content • files/en-us/web/api/element/setcapture/index.html:
  79: <p><a href="/samples/domref/mousecapture.html">View Live Examples</a></p>

content • files/en-us/web/api/fullscreen_api/index.html:
  134: <p><a href="/samples/domref/fullscreen.html">View Live Examples</a></p>

content • files/en-us/web/api/fullscreen_api/guide/index.html:
  91: <p><a href="/samples/domref/fullscreen.html">View Live Examples</a></p>

content • files/en-us/web/api/window/showmodaldialog/index.html:
  74: <p><a href="/samples/domref/showModalDialog.html">Try out <code>showModalDialog()</code></a>.</p>

content • files/en-us/web/css/alternative_style_sheets/index.html:
  35: <p><a href="/samples/cssref/altstyles/index.html">Click here</a> for a working example you can try out.</p>

content • files/en-us/web/css/element/index.html:
  34: <p>These examples can be <a href="/samples/cssref/moz-element.html" title="https://developer.mozilla.org/samples/cssref/moz-element.html">viewed live</a> in builds of Firefox that support <code>-moz-element()</code>.</p>

content • files/en-us/web/demos_of_open_web_technologies/index.html:
  26:  <li><a href="/samples/raycaster/RayCaster.html" title="https://developer.mozilla.org/editor/fckeditor/core/editor/samples/raycaster/RayCaster.html">3D RayCaster</a></li>

content • files/en-us/web/guide/printing/index.html:
  68: <div><a href="/samples/domref/printevents.html">View Live Examples</a></div>
```

Reply:
Thanks. That line was converted right after the first meeting we had about this and I totally forgot.

Anyway. Looking at it I like just using https://media.prod.mdn.mozit.cloud/  - keeps things simple.

One thing I observed is that https://media.prod.mdn.mozit.cloud/* urls seem to be case sensitive which is an issue as we lose case very early.
https://media.prod.mdn.mozit.cloud/samples/html/currentScript.html vs https://media.prod.mdn.mozit.cloud/samples/html/currentscript.html

This should work for the Lambda@Edge though and I'll try to think of something for the other cases / if it matters there.